parmed/openmm/parameters.py

```diff
@@ -178,10 +179,16 @@ def _write_omm_atom_types(self, dest):
         dest.write(' </AtomTypes>\n')
 
     def _write_omm_residues(self, dest):
+        if not self.residues: return
         dest.write(' <Residues>\n')
         for name, residue in iteritems(self.residues):
             if not isinstance(residue, ResidueTemplate):
                 continue
+            # if AtomTypes are available, this will throw an exception if the
+            # residue is claiming a Type that doesn't exist
+            if self.atom_types:
```

Comment:
> Ah, I see from the commit message. Maybe we can make typeify_templates a little smarter about only "typeifying" what it can and not complaining when some types don't exist?

The thing is I don't see the point of 'typeifying' other than it being a test for the existance of the atom types - because all it does is bind the AtomType objects to `atom.atom_type` but we subsequently don't use that at all - the writing function uses `atom.type`. So making it not complain when some types don't exist makes it essentially pointless, right?

> Also, maybe we should get rid of ResidueTemplateContainer instances (basically the solvent boxes) when calling OpenMMParameterSet.from_parameterset so that we only keep residues that we explicitly plan on writing to XML files. What do you think about that?

I don't see much difference between removing these at calling OpenMMParameterSet.from_parameterset or at writing residues to be fair - but maybe that would make the code a bit cleaner. Would there be any use case for having the Containers part of the OpenMMParameterSet in the future? 

How about we do:
* have the 'self.typeify_templates' call but add a 'if not self.atom_types: return` line to the `typeify_templates` function (so if no Atom Types available - assume user takes the risk of some atom types possibly being wrong, if any Atom Types available - assume user wants the check). But if feeding the `atom.atom_type` 's is useless like I think - maybe we should just replace that with an `assert` test? @ChayaSt how does this work for you? She has the stream files, and it seems like they might contain some Atom Types but not all? In this case this wouldn't work and we should maybe just get rid of typeify_templates completely and accept the risk? 

* and remove the `ResidueTemplateContainer` instances at calling `OpenMMParameterSet.from_parameterset`

How does this sound?